### PR TITLE
Removes allocation versioning

### PR DESCRIPTION
**Pull request by willfish**
### Jira link

P4-<TODO>

### What?

I have added/removed/altered:

- [x] Removes versioning in allocations endpoints

### Why?

I am doing this because:

- This versioning had nothing to do with serializers and was only related to whether we supported default includes or not
- Since only the frontend uses this and it's on v2, let's just remove versioning.

### Have you? (optional)

- [x] Updated API docs if necessary

### CODE REVIEW

**Comment by willfish** [comment]
We never documented the version difference between v1/v2 allocation includes so nothing to change in swagger docs.

File: app/controllers/api/v1/allocations_actions.rb [hunk 1]
```diff
@@ -1,7 +0,0 @@
-module Api::V1
-  module AllocationsActions
-    def included_relationships
-      IncludeParamHandler.new(params).call || AllocationSerializer::SUPPORTED_RELATIONSHIPS
```
Comment (smoothcontract) [inline app/controllers/api/v1/allocations_actions.rb:4]: Argh!! This is the file that caused us so much grief when testing - it defaults to returning all of the includes if none were specified. Kill it now. Kill it with fire!

**Review by smoothcontract — APPROVED** [review]
So nice to see this cleaned up and V2 behaviour get promoted 👏

File: app/controllers/api/v1/allocations_actions.rb (hunk 1 above)
Comment (willfish) [reply app/controllers/api/v1/allocations_actions.rb:4]: 💥

**Pull request merged**